### DISCUSSION TITLE
Questions about the coefficients in the MC of coverage_map

### DISCUSSION THREAD

**Discussion by liyuxuan5603** [NVlabs/sionna#628] [Q&A] [answered]
I am very interested in the Monte Carlo process you've implemented, but I find myself puzzled by the coefficients, particularly after reviewing the code and struggling to find a clear correspondence. I have two questions: 
First, can the values calculated in coverage_map be effectively approximated as the power at the cell centers? 
Second, could you explain how the coefficients in the code align with the formulas? 
Thank you for your help.
![image](https://github.com/user-attachments/assets/50a2b67a-6aac-44f0-812f-4053ff371844)

![微信图片_20241024165651](https://github.com/user-attachments/assets/89bd1914-79ef-4b4d-8b9b-43462e160265)

**Comment by faycalaa** [comment, marked as answer]
Hi @liyuxuan5603,
1. Assuming that the number of samples used to compute the coverage map is sufficiently large and that the cell size is small enough, then the path loss measured by a receiver at the center of the coverage map cells should match the ones of the coverage map. [Note that some of the unit tests of Sionna precisely check this.](https://github.com/NVlabs/sionna/blob/0aeca346148f1e47a4303f391da1db07e6df1ea7/test/unit/rt/test_coverage_map.py#L575).
2. The scaling in the code you have highlighted combines the scaling from Eq. (53) you mentioned and [the scaling from the computation of channel impulse responses](https://nvlabs.github.io/sionna/em_primer.html#equation-h-final).

**Reply by liyuxuan5603** [reply]
@faycalaa I'm still confused with the term $r(\omega_n)^2$, could you point out the corresponding implementation in the code? Thank you for your kind help.

**Reply by faycalaa** [reply]
The term $r(\omega_n)^2$ cancels out when considering free space propagation ($E_r(\omega_n) \propto \frac{1}{r(\omega_n)}$, where $E_r$ is the received electric field). This is implemented by applying neither the free space propagation loss nor the scaling by $r(\omega_n)^2$ in Eq. (53) you mentioned.

**Reply by liyuxuan5603** [reply]
I've totally understood the equation. Thank you for your patient answer.
